### PR TITLE
Remove metrics

### LINKED ISSUE #3732: Remove Metrics API
### Description

Remove Metrics API since it will No-Op in the product soon.

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Remove metrics API and internal classes.

Metrics will return in the future but we don't know in what form exactly.

## :bulb: Motivation and Context

Closes #3732 

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Review by markushi — APPROVED** [review]
🫡 nice one! I did a `grep` on top and couldn't find any remaining leftovers.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 ### Breaking Changes
 
 - Throw IllegalArgumentException when calling Sentry.init on Android ([#3596](https://github.com/getsentry/sentry-java/pull/3596))
+- Metrics has been removed from the SDK ([#3774](https://github.com/getsentry/sentry-java/pull/3774))
```
Comment (adinauer) [inline CHANGELOG.md:8]:
```suggestion
- Metrics have been removed from the SDK ([#3774](https://github.com/getsentry/sentry-java/pull/3774))
```

**Comment by sysmat** [comment]
are any alternatives for `Sentry.metrics()`?

**Comment by adinauer** [comment]
@sysmat we're still figuring out how it'll work in the future, can't say when we have something to share

**Pull request merged**